definitions.h

```diff
@@ -47,6 +52,327 @@ struct Vertex
     double w;
 };
 
+class Matrix
+{
+    public:
+		Matrix();
+        Matrix(int c, int r);
+        void insert(int x, int y, double data);
+        double returnVal(int x, int y);
+		int getColumnSize();
+		int getRowSize();
+		int getSize();
+		//Matrix matrixCopy(Matrix & m);
+		//Have to write 4 different const types
+		Matrix operator*(const Matrix & m) const;
+		//Matrix operator*(const Vertex & m) const;
+		Vertex operator*(const Vertex & m) const;
+		Matrix& operator=(Matrix const& m);
+		Matrix & operator=(std::initializer_list<double> m);
+		Matrix & operator=(std::initializer_list<std::initializer_list<double>> m);
+
+    private:
+        int column;
+        int row;
+		int size;
+        double matrixData[16];

```

Comment:
I see you used a 1D array here.  I can see how you implemented your Matrix class.  Good job in using C++ constructs to do this.